src/core/metadata/qgslayermetadata.cpp

```diff
@@ -273,6 +278,312 @@ void QgsLayerMetadata::readFromLayer( const QgsMapLayer *layer )
   mLinks = layer->customProperty( QStringLiteral( "metadata/links" ) ).value<LinkList>();
 }
 
+bool QgsLayerMetadata::readMetadataXml( const QDomElement &metadataElement )
+{
+  QDomNode mnl;
+  QDomElement mne;
+
+  // set identifier
+  mnl = metadataElement.namedItem( QStringLiteral( "identifier" ) );
+  mIdentifier = mnl.toElement().text();
+
+  // set parent identifier
+  mnl = metadataElement.namedItem( QStringLiteral( "parentidentifier" ) );
+  mParentIdentifier = mnl.toElement().text();
+
+  // set language
+  mnl = metadataElement.namedItem( QStringLiteral( "language" ) );
+  mLanguage = mnl.toElement().text();
+
+  // set type
+  mnl = metadataElement.namedItem( QStringLiteral( "type" ) );
+  mType = mnl.toElement().text();
+
+  // set title
+  mnl = metadataElement.namedItem( QStringLiteral( "title" ) );
+  mTitle = mnl.toElement().text();
+
+  // set abstract
+  mnl = metadataElement.namedItem( QStringLiteral( "abstract" ) );
+  mAbstract = mnl.toElement().text();
+
+  // set keywords
+  QDomNodeList keywords = metadataElement.elementsByTagName( QStringLiteral( "keywords" ) );
+  for ( int i = 0; i < keywords.size(); i++ )
+  {
+    QStringList keywordsList;
+    mnl = keywords.at( i );
+    mne = mnl.toElement();
+
+    QDomNodeList el = mne.elementsByTagName( QStringLiteral( "keyword" ) );
+    for ( int j = 0; j < el.size(); j++ )
+    {
+      keywordsList.append( el.at( j ).toElement().text() );
+    }
+    addKeywords( mne.attribute( QStringLiteral( "vocabulary" ) ), keywordsList );
+  }
+
+  // set fees
+  mnl = metadataElement.namedItem( QStringLiteral( "fees" ) );
+  mFees = mnl.toElement().text();
+
+  // constraints
+  QDomNodeList constraintsList = metadataElement.elementsByTagName( QStringLiteral( "constraints" ) );
+  for ( int i = 0; i < constraintsList.size(); i++ )
+  {
+    mnl = constraintsList.at( i );
+    mne = mnl.toElement();
+    addConstraint( QgsLayerMetadata::Constraint( mne.text(), mne.attribute( QStringLiteral( "type" ) ) ) );
```

Comment:
Same here - you'll need to clear any existing constraints first

Reply:
Done